lib/pbench/server/database/models/users.py

```diff
@@ -91,24 +98,38 @@ def evaluate_role(self, key: str, value: str) -> Optional[str]:
         except KeyError:
             return None
 
-    @validates("password")
-    def evaluate_password(self, key: str, value: str) -> str:
-        return generate_password_hash(value)
-
     # validate the email field
     @validates("email")
     def evaluate_email(self, key: str, value: str) -> str:
         valid = validate_email(value)
         return valid.email
 
+    def updateDict(self, orig_dict: JSON, new_dict: JSON) -> JSON:
+        """Update the Original json dict with a new json dict
+        Args:
+            orig_dict: Original JSON object to update
+            new_dict: New JSON object to parse
+        Returns:
+            Update JSON object
+        """
+        for key, val in new_dict.items():
+            if isinstance(val, dict):
+                tmp = update(orig_dict.get(key, {}), val)
+                orig_dict[key] = tmp
+            elif isinstance(val, list):
```

Comment:
If someone passes in a `list` you're basically assuming (but not checking) that the *existing* key value is also a `list`; and the same for `dict`. Since you're updating, you need to make sure the actual value structure is compatible with the new value structure.

I'd like to see the docstring expanded with a vision of exactly how you see this being used to manage a "profile". You might want to consider something like the `server_settings` validator structure that enforces specific semantics for predefined keys if you have any in mind. And if it's just freeform like the metadata `global` and `user`, you might consider a reserved and managed namespace as well that the *server* can potentially use. (For example, your first name, last name, email properties?)

Reply:
> If someone passes in a `list` you're basically assuming (but not checking) that the _existing_ key value is also a `list`; and the same for `dict`. Since you're updating, you need to make sure the actual value structure is compatible with the new value structure.

Yes, I should fix that.

> I'd like to see the docstring expanded with a vision of exactly how you see this being used to manage a "profile". You might want to consider something like the `server_settings` validator structure that enforces specific semantics for predefined keys if you have any in mind. And if it's just freeform like the metadata `global` and `user`, you might consider a reserved and managed namespace as well that the _server_ can potentially use. (For example, your first name, last name, email properties?)

Ultimately I would hope to have this managed in the same way as we manage the metadata keys. Users can specify the dot-separated keys that they want to put as profile keys and the server should unflatten and validate those dot-separated keys to update the original profile value. We could have some freeform keys that Users are free to use and some are only updatable by the server etc. 

However, this is something I am not thinking too much into right now like what kind of keys should be freeform and what should be server reserved. That will come later as we understand what goes in the profile column and what we should allow in this column. We haven't had a discussion around that yet so I was thinking to keep it simple for now and get this model working before I make any enhancements to the Users table.

Reply:
I have addressed it in the latest commit. I am only keeping internal `id`, `oidc_id`, and `username` and everything else that we extract from the token will go to the `profile` key.